### PR TITLE
[MM-38145] Use mattermost-server/v6

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR updates the dependency on `github.com/mattermost/mattermost-server` from `v5` to `v6`. This ensures the v6 breaking changes propagate quickly to developers using this plugin.

Please note that the plugin still works against v5 servers.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38145

### CODE REVIEW

**Comment by streamer45** [comment]
@hanzei  My only worry here (and related PRs) is that we haven't yet merged all the planned breaking changes so wondering if we should wait until that's the case to avoid doing extra work.

**Comment by hanzei** [comment]
Good point @streamer45. I don't think any of the upcoming changes are effecting plugins. Still, maybe we keep this PR open until the first rc has been cut (tomorrow) and update the tag. How does that sound?

**Comment by streamer45** [comment]
Sounds good to me. Thanks :+1:

**Comment by streamer45** [comment]
@hanzei PR was merged so we should be good to update.

**Comment by hanzei** [comment]
@streamer45 @mickmister Please take another look

**Review by mickmister — APPROVED** [review]

File: build/go.mod [hunk 1]
```diff
@@ -3,6 +3,6 @@ module github.com/mattermost/mattermost-plugin-starter-template/build
 go 1.12
 
 require (
-	github.com/mattermost/mattermost-server/v5 v5.3.2-0.20200723144633-ed34468996e6
+	github.com/mattermost/mattermost-server/v6 v6.0.0-20210824183758-eaa37b52ee5a
```
Comment (streamer45) [inline build/go.mod:6]: Isn't this commit prior to the latest changes?

Comment (hanzei) [reply build/go.mod]: Good catch :+1: Fixed.

**Review by streamer45 — APPROVED** [review]
Nicely done :+1:

**Comment by hanzei** [comment]
@DHaussermann Same comment as in https://github.com/mattermost/mattermost-plugin-autolink/pull/169#issuecomment-911295678 :wink:

**Comment by hanzei** [comment]
@DHaussermann I'm going to defer the QA approval to release testing.

**Pull request merged**